api/client/inspect/inspector.go

```diff
@@ -70,26 +71,49 @@ func NewIndentedInspector(outputStream io.Writer) Inspector {
 }
 
 // Inspect writes the raw element with an indented json format.
-func (i *IndentedInspector) Inspect(typedElement interface{}, _ []byte) error {
-	i.elements = append(i.elements, typedElement)
+func (i *IndentedInspector) Inspect(typedElement interface{}, rawElement []byte) error {
+	if rawElement != nil {
+		i.rawElements = append(i.rawElements, rawElement)
+	} else {
+		i.elements = append(i.elements, typedElement)
+	}
 	return nil
 }
 
 // Flush write the result of inspecting all elements into the output stream.
 func (i *IndentedInspector) Flush() error {
-	if len(i.elements) == 0 {
+	if len(i.elements) == 0 && len(i.rawElements) == 0 {
 		_, err := io.WriteString(i.outputStream, "[]\n")
 		return err
 	}
 
-	buffer, err := json.MarshalIndent(i.elements, "", "    ")
-	if err != nil {
-		return err
+	var buffer io.Reader
+	if len(i.rawElements) > 0 {
```

Comment:
This is intentional, the elements are the same, but in two different representations. We won't have both ever and we're favoriting the raw bytes because the typed elements can have missing fields, like it's the case of containers coming from swarm with the `Node` structure.